### PR TITLE
fix: delete duplicated trainee profile placements caused by missing ID.

### PR DESCRIPTION
**Pull request by DorisWongNHS**
TIS21-3171

### CODE REVIEW

File: src/test/java/uk/nhs/hee/trainee/details/migration/DeleteDuplicateTraineeProfilePlacementsTest.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        eq(PROFILE_COLLECTION)))
+        .thenReturn(result);
+    when(result.getModifiedCount()).thenReturn(1L);
+
+    migration.migrate();
+
+    Query queryUsed = queryCaptor.getValue();
+    assertTrue("Unexpected query.", queryUsed.getQueryObject().isEmpty());
+    Update updateUsed = updateCaptor.getValue();
+    assertTrue("Unexpected placement deletion", updateUsed.equals(update));
```
Comment (Judge40) [inline src/test/java/uk/nhs/hee/trainee/details/migration/DeleteDuplicateTraineeProfilePlacementsTest.java:71]: I'd rather we used Hamcrest matchers here as they usually generate better failure messages.

**Comment by Judge40** [comment]
We could add a test that the MongoException is caught rather than being thrown.

File: src/test/java/uk/nhs/hee/trainee/details/migration/DeleteDuplicateTraineeProfilePlacementsTest.java (hunk 1 above)
Comment (DorisWongNHS) [reply src/test/java/uk/nhs/hee/trainee/details/migration/DeleteDuplicateTraineeProfilePlacementsTest.java]: Thanks, it has been updated to use Hamcrest matchers.

**Review by ReubenRobertsHEE — APPROVED** [review]
LGTM :)

**Pull request merged**